### DISCUSSION TITLE
Unique (user-configured) linenum anchors for Code Blocks

### DISCUSSION THREAD

**Discussion by hellt** [facelessuser/pymdown-extensions#2258] [Q&A] [answered]
Hi all,
I am in love with the way one can link to certain line numbers from the markdown:

```
Here we [handle the `version`](#__codelineno-8-2:9) CLI flag.
```

The problem I am having is that `codelineno-8-2:9` is not a fixed id of the element.
If we add a code block before the one in question, its ID will auto-increment and the link won't work anymore.

It would be cool to be use either random ID to avoid clashes, or let the user to define it.

**Comment by facelessuser** [comment, marked as answer]
> It would be cool to be use either random ID to avoid clashes, or let the user to define it.

Static IDs are already possible, just attach an ID to your code block:

````
``` {.python #myid linenums="1"}
import bar
```
````

```html
<div id="myid" class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__codeline-myid-1"><span class="kn">import</span> <span class="nn">bar</span>
</span></code></pre></div></td></tr></table></div>
```

**Reply by hellt** [reply]
so it seems the problem is on the mkdocs-material side then, I see the IDs assigned to span elements, but the hrefs to the manually assigned IDs do not lead to highlighting the ranges in the code blocks.

https://github.com/squidfunk/mkdocs-material/discussions/6415
